packages/components/src/internal/components/picklist/actions.ts

```diff
@@ -153,38 +178,167 @@ export function updatePicklist(picklist: PicklistModel): Promise<PicklistModel>
     });
 }
 
+export interface SampleTypeCount {
+    ItemCount: number;
+    SampleType: string;
+    LabelColor: string;
+}
+
+export function getPicklistCountsBySampleType(listName: string): Promise<SampleTypeCount[]> {
+    return new Promise((resolve, reject) => {
+        Query.executeSql({
+            sql:
+                'SELECT COUNT(*) as ItemCount, \n' +
+                '       SampleId.SampleSet.Name AS SampleType, \n' +
+                '       SampleId.LabelColor\n' +
+                'FROM lists."' +
+                listName +
+                '"\n' +
+                'GROUP BY SampleId.SampleSet.Name, SampleId.LabelColor\n' +
+                'ORDER BY SampleId.SampleSet.Name',
+            schemaName: 'lists',
+            success: data => {
+                resolve(data.rows);
+            },
+            failure: reason => {
+                console.error(reason);
+                reject(reason);
+            },
+        });
+    });
+}
+
+export function getPicklistSamples(listName: string): Promise<Set<string>> {
+    return new Promise((resolve, reject) => {
+        const schemaName = 'lists';
+        selectRows({
+            schemaName,
+            queryName: listName,
+        })
+            .then(response => {
+                const {models} = response;
+                const dataKey = resolveKey(schemaName, listName);
+                resolve(new Set(Object.values(models[dataKey]).map((row: any) => row.SampleID.value.toString())));
+            })
+            .catch(reason => {
+                console.error(reason);
+                reject(reason);
+            });
+    });
+}
+
+export function getSelectedPicklistSamples(
+    picklistName: string,
+    selectedIds: string[],
+    saveSnapshot?: boolean,
+    selectionKey?: string
+): Promise<number[]> {
+    return new Promise((resolve, reject) => {
+        getSelectedData(
+            SCHEMAS.PICKLIST_TABLES.SCHEMA,
+            picklistName,
+            selectedIds,
+            [PICKLIST_SAMPLE_ID_COLUMN, PICKLIST_KEY_COLUMN].join(','),
+            undefined,
+            undefined,
+            PICKLIST_KEY_COLUMN
+        )
+            .then(response => {
+                const {data} = response;
+                const sampleIds = [];
+                const rowIds = [];
+                data.forEach(row => {
+                    sampleIds.push(row.getIn([PICKLIST_SAMPLE_ID_COLUMN, 'value']));
+                    if (saveSnapshot) {
+                        rowIds.push(row.getIn([PICKLIST_KEY_COLUMN, 'value']));
+                    }
```

Comment:
you likely already checked this, but do we need any sort of case sensitivity checks on these row prop value access calls?